File: Solutions/1422. Maximum Score After Splitting a String/1422.java
class Solution {
    public int maxScore(String s) {
        int left=0, right =0;
        left = s.charAt(0) == '0' ? 1 : 0;
        for(int i=1;i<s.length();i++){
            if(s.charAt(i) == '1'){
                right++;
            }
        }
        int max = left + right;
        for(int i=1;i<s.length() - 1;i++){
            if(s.charAt(i) == '0'){
                left++;
            } else{
                right--;
            }
            max = Math.max(max, left+right);
        }
        return max;
    }
}

File: Solutions/1422. Maximum Score After Splitting a String/1422.py
class Solution:
    def maxScore(self, s: str) -> int:
        left = 1 if s[0] == '0' else 0
        right = 0

        for i in range(1, len(s)):
            if s[i] == '1':
                right += 1

        ans = right + left

        for i in range(1, len(s) - 1):
            if s[i] == '0':
                left += 1
            else:
                right -= 1
            ans = max(ans, left + right)
        
        return ans